test/strategies/kandel/abstract/CoreKandel.gas.t.sol

```diff
@@ -0,0 +1,164 @@
+    setupMarket(base, quote);
+  }
+
+  function densifyMissing(uint index, uint fold) internal {
+    IndexStatus memory idx = getStatus(index);
+    if (idx.status == OfferStatus.Bid) {
+      // densify Ask position
+      densify(address(base), address(quote), idx.bid.gives(), idx.bid.wants(), 0, 0, fold, address(this));
+    } else {
+      if (idx.status == OfferStatus.Ask) {
+        densify(address(quote), address(base), idx.ask.gives(), idx.ask.wants(), 0, 0, fold, address(this));
+      }
+    }
+  }
+
+  function test_log_mgv_config() public view {
+    (, MgvStructs.LocalPacked local) = mgv.config($(base), $(quote));
+    console.log("offer_gasbase", local.offer_gasbase());
+    console.log("kandel gasreq", kdl.offerGasreq());
+  }
+
+  function test_complete_fill_bid_order() public {
+    uint completeFill = completeFill_;
+    _gas();
+    vm.prank(taker);
+    // taking partial fill to have gas cost of reposting
+    (uint takerGot,,,) = mgv.marketOrder($(base), $(quote), completeFill, type(uint160).max, true);
+    gas_();
+    require(takerGot > 0);
+  }
+
+  function test_bid_order_length_1() public {
+    uint partialFill = partialFill_;
+    _gas();
+    vm.prank(taker);
+    // taking partial fill to have gas cost of reposting (all offers give 0.108 ethers)
+    (uint takerGot,,,) = mgv.marketOrder($(base), $(quote), partialFill, type(uint160).max, true);
+    uint g = gas_(true);
+    assertEq(reader.minusFee($(base), $(quote), partialFill), takerGot, "Incorrect got");
+    console.log(1, ",", g);
+    assertStatus(4, OfferStatus.Bid);
+  }
+
+  function test_bid_order_length_2() public {
+    uint completeFill = completeFill_;
+    uint partialFill = partialFill_;
+    _gas();
+    vm.prank(taker);
+    (uint takerGot,,,) = mgv.marketOrder($(base), $(quote), completeFill + partialFill, type(uint160).max, true);
+    uint g = gas_(true);
+    require(takerGot > 0);
+    console.log(2, ",", g);
+    assertStatus(3, OfferStatus.Bid);
+  }
+
+  function test_bid_order_length_3() public {
+    uint completeFill = completeFill_;
+    uint partialFill = partialFill_;
+    _gas();
+    vm.prank(taker);
+    (uint takerGot,,,) = mgv.marketOrder($(base), $(quote), completeFill * 2 + partialFill, type(uint160).max, true);
+    uint g = gas_(true);
+    require(takerGot > 0);
+    console.log(3, ",", g);
+    assertStatus(2, OfferStatus.Bid);
+  }
+
+  function test_bid_order_length_4() public {
+    uint completeFill = completeFill_;
+    uint partialFill = partialFill_;
+    _gas();
+    vm.prank(taker);
+    (uint takerGot,,,) = mgv.marketOrder($(base), $(quote), completeFill * 3 + partialFill, type(uint160).max, true);
+    uint g = gas_(true);
+    require(takerGot > 0);
+    console.log(4, ",", g);
+    assertStatus(1, OfferStatus.Bid);
+  }
+
+  function test_bid_order_length_5() public {
+    uint completeFill = completeFill_;
+    uint partialFill = partialFill_;
+    _gas();
+    vm.prank(taker);
```

Comment:
In test_offerLogic_partialFill_cost it seems you can invoke _gas() before prank - shouldn't that be done in every case?

Reply:
corrected (although I don't think prank consumes gas)